### PR TITLE
[R20-292] - Preview page

### PR DESCRIPTION
**Pull request by jeffdowdle**
**Issue**:

### What I did

- 
- 

### How to test

- 
- 

#### For all PR's

- [ ] I've added relevant changes to the project Readme if needed.
- [ ] I've updated the documentation site as needed.
- [ ] I have added unit tests to cover my changes (if not applicable, please state why in a comment)

#### For new components only

- [ ] I have added a story covering all variants
- [ ] I have checked a11y tab in storybook passes
- [ ] Any events are emitted on the event bus

### CODE REVIEW

File: packages/nuxt-ripple-preview/server/plugins/preview.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  const tideBaseUrl = config.public.tide.baseUrl
+
+  const authClient = new ClientOAuth2({
+    clientId: config.tide.preview.clientId,
+    clientSecret: config.tide.preview.clientSecret,
+    accessTokenUri: `${tideBaseUrl}/oauth/token`,
+    authorizationUri: `${tideBaseUrl}/oauth/authorize`,
+    redirectUri: 'http://localhost:3000/oauth/login',
```
Comment (jeffdowdle) [inline packages/nuxt-ripple-preview/server/plugins/preview.ts:33]: I was unable to figure out how to get the current site base url from within this server plugin.

File: packages/nuxt-ripple-preview/server/routes/oauth/login.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * Is contains the code and state values in the query string, which are used to get an access token.
+ *
+ * We must validate here that state value is the same as the one sent during the initial login request to mitigate CSRF attacks.
+ *
+ * The access token and expiry is then stored in a cookie and the user is redirected to the success page.
+ */
+export const createPageHandler = async (event: H3Event, authClient) => {
+  return createHandler(event, 'TideOAuthLoginHandler', async () => {
+    // TODO: Validate state value before getting token
```
Comment (jeffdowdle) [inline packages/nuxt-ripple-preview/server/routes/oauth/login.ts:17]: TODO

File: packages/nuxt-ripple-preview/server/api/tide/oauth.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export const createPageHandler = async (
+  event: H3Event,
+  authClient: ClientOAuth2
+) => {
+  return createHandler(event, 'TideOAuthHandler', async () => {
+    // Create a random state value to prevent CSRF attacks, this should be unique per request
+    const state = nanoId()
+
+    // TODO - Store the state value in a cookie or session
```
Comment (jeffdowdle) [inline packages/nuxt-ripple-preview/server/api/tide/oauth.ts:16]: TODO

File: .vscode/settings.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
     "vue-app",
     "@dpc-sdp/ripple-tide-search",
     "@dpc-sdp/ripple-tide-event",
+    "@dpc-sdp/ripple-tide-preview",
```
Comment (lambry) [inline .vscode/settings.json:22]: Hey @jeffdowdle, I'm guessing this can be changed to `@dpc-sdp/nuxt-ripple-preview`.

**Review by waitingallday — COMMENTED** [review]
Looks good, it all works! I'm seeing this at the start on first run of nuxt-app (dev) though

```
WARN  Duplicated imports "AuthCookieNames", the one from "/Users/jason.smith/rep/_cleanroom/ripple-framework/packages/nuxt-ripple-preview/utils/constants.ts" has been ignored

WARN  Duplicated imports "isPreviewPath", the one from "/Users/jason.smith/rep/_cleanroom/ripple-framework/packages/nuxt-ripple-preview/utils/index.ts" has been ignored
```

**Review by lambry — APPROVED** [review]

**Review by waitingallday — APPROVED** [review]

**Review by dylankelly — APPROVED** [review]
great job on this

**Pull request merged**